### DISCUSSION TITLE
Best way to destroy textbox?

### DISCUSSION THREAD

**Discussion by kristajg** [rexrainbow/phaser3-rex-notes#264] [Q&A] [answered]
Hi, simple question for you all:

I'm using the [UI-TextBox plugin](https://rexrainbow.github.io/phaser3-rex-notes/docs/site/ui-textbox/).
It's working great to initialize it, but what is the best way to destroy this AKA remove it from my scene?

**Comment by rexrainbow** [comment, marked as answer]
[`textBox.destroy()`](https://github.com/rexrainbow/phaser3-rex-notes/blob/master/examples/ui-textbox/destroy.js#L28)

**Reply by kristajg** [reply]
Hey @rexrainbow Thanks for the quick reply. I've been trying every way I can to get this method to work on textBox, but it simply isn't working. Below is a gist of my latest setup, could you please tell me what I'm doing wrong?

https://gist.github.com/kristajg/ccda40301774c2ea07ce17a58f8199cc

**Reply by rexrainbow** [reply]
Please ensure that [textBox](https://gist.github.com/kristajg/ccda40301774c2ea07ce17a58f8199cc#file-phasertextboxexample-js-L97) is created one time, before destroying.

**Reply by kristajg** [reply]
It is being created, could you perhaps clarify? Maybe I'm missing something simple

- this.textBox variable is initialized in preloader
- createTextBox function runs in the line you reference, assigning the result to the variable
- The text box does show up in the ui
- When I run destroy() on it, nothing happens. No error, but it doesn't go away either

I've also tried initializing textBox as a plain variable above the class for a broader scope, but that didnt help either

**Reply by rexrainbow** [reply]
I guess that `createTextBox()` method had been invoked many times, but only last one instance will be assigned to `this.textBox`. And `this.textBox.destroy()` only destroy latest one.

**Reply by kristajg** [reply]
You are right, thank you!
